http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClientBuilder.java

```diff
@@ -0,0 +1,268 @@
+        return this;
+    }
+
+    /**
+     * Set the optional URI for this client to use as the root.
+     *
+     * @param uri The URI
+     * @return This builder
+     */
+    @NonNull
+    public DefaultHttpClientBuilder uri(@Nullable URI uri) {
+        return loadBalancer(uri == null ? null : LoadBalancer.fixed(uri));
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder explicitHttpVersion(@Nullable HttpVersionSelection explicitHttpVersion) {
+        this.explicitHttpVersion = explicitHttpVersion;
+        return this;
+    }
+
+    /**
+     * Set the configuration.
+     *
+     * @param configuration The client configuration
+     * @return This builder
+     */
+    @NonNull
+    public DefaultHttpClientBuilder configuration(@NonNull HttpClientConfiguration configuration) {
+        ArgumentUtils.requireNonNull("configuration", configuration);
+        this.configuration = configuration;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder contextPath(@Nullable String contextPath) {
+        this.contextPath = contextPath;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder filterResolver(@NonNull HttpClientFilterResolver<ClientFilterResolutionContext> filterResolver) {
+        ArgumentUtils.requireNonNull("filterResolver", filterResolver);
+        this.filterResolver = filterResolver;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder annotationMetadataResolver(@Nullable AnnotationMetadataResolver annotationMetadataResolver) {
+        if (annotationMetadataResolver != null) {
+            this.annotationMetadataResolver = annotationMetadataResolver;
+        }
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder filters(HttpClientFilter... filters) {
+        return filterResolver(new DefaultHttpClientFilterResolver(null, annotationMetadataResolver, Arrays.asList(filters)));
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder clientFilterEntries(@Nullable List<HttpFilterResolver.FilterEntry> clientFilterEntries) {
+        this.clientFilterEntries = clientFilterEntries;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder threadFactory(@Nullable ThreadFactory threadFactory) {
+        this.threadFactory = threadFactory;
+        return this;
+    }
+
+    /**
+     * The netty SSL context builder. Used by the micronaut-oracle-cloud OKE workload identity
+     * client.
+     *
+     * @param nettyClientSslBuilder The SSL context builder
+     * @return This builder
+     */
+    @NonNull
+    public DefaultHttpClientBuilder nettyClientSslBuilder(@NonNull ClientSslBuilder nettyClientSslBuilder) {
+        ArgumentUtils.requireNonNull("nettyClientSslBuilder", nettyClientSslBuilder);
+        this.nettyClientSslBuilder = nettyClientSslBuilder;
+        return this;
+    }
+
+    @NonNull
+    DefaultHttpClientBuilder codecRegistry(@NonNull MediaTypeCodecRegistry codecRegistry) {
```

Comment:
Please already deprecate

Reply:
Done. In fact this is part of the motivation for the builder. [In the oracle cloud module](https://github.com/micronaut-projects/micronaut-oracle-cloud/blob/8c60ad850afd16e94ac71585332f2507794811d3/oraclecloud-oke-workload-identity/src/main/java/io/micronaut/oraclecloud/oke/workload/identity/MicronautOkeWorkloadIdentityResourcePrincipalsFederationClient.java#L163), there is code that has to construct this registry because the constructor it uses requires it, even though the defaults would be fine. With this change, that code will only have to set the configuration and ssl builder, leaving the other builder properties as default, and we can change or remove them in core without breaking things.